app/src/main/java/com/github/se/polyfit/ui/components/selector/MealOccasionSelector.kt

```diff
@@ -0,0 +1,96 @@
+package com.github.se.polyfit.ui.components.selector
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.RadioButton
+import androidx.compose.material3.RadioButtonColors
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.github.se.polyfit.model.meal.MealOccasion
+import com.github.se.polyfit.ui.theme.PrimaryPurple
+import com.github.se.polyfit.ui.theme.PurpleGrey40
+import com.github.se.polyfit.ui.utils.titleCase
+
+@Composable
+fun MealOccasionSelector(
+    startOccasion: MealOccasion,
+    onConfirm: (MealOccasion) -> Unit,
```

Comment:
Code looks amazing ! If I  understood correctly the `onConfirm` function, will update the meal in the `mealViewModel` every time the user clicks on a new occasion ?

Reply:
That's what it'll be used for yeah :D But to make it flexible, instead of having to pass a mealviewmodel you can give it any callback to update whatever you want